lib/HLSL/ComputeViewIdStateBuilder.cpp

```diff
@@ -481,17 +486,23 @@ void DxilViewIdStateBuilder::CollectValuesContributingToOutputs(
       GetUnsignedVal(SPO.get_outputSigId(), &id);
       GetUnsignedVal(SPO.get_colIndex(), &col);
       GetUnsignedVal(SPO.get_rowIndex(), (uint32_t *)&startRow);
+      IsForPatchConstantOrPrimitive = true;
     } else if (DxilInst_StorePatchConstant SPC =
                    DxilInst_StorePatchConstant(CI)) {
       pDxilSig = &m_pModule->GetPatchConstOrPrimSignature();
       pContributingValue = SPC.get_value();
       GetUnsignedVal(SPC.get_outputSigID(), &id);
       GetUnsignedVal(SPC.get_row(), (uint32_t *)&startRow);
       GetUnsignedVal(SPC.get_col(), &col);
+      IsForPatchConstantOrPrimitive = true;
     } else {
       IFT(DXC_E_GENERAL_INTERNAL_ERROR);
     }
 
+    // Skip vertex outputs for MS when check primitive outputs.
```

Comment:
I don't quite understand why these flags being different is the way to determine if we're checking for primitive outputs or not?

Is there not a more direct "if (CheckingPrimitiveOutputs)" type thing we could be saying here?

Reply:
Switched to check IsForPatchConstantOrPrimitive.